src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -126,7 +126,11 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
 
                 // Sort and validate graph
                 Workflow provisionWorkflow = template.workflows().get(PROVISION_WORKFLOW);
-                List<ProcessNode> provisionProcessSequence = workflowProcessSorter.sortProcessNodes(provisionWorkflow, workflowId);
+                List<ProcessNode> provisionProcessSequence = workflowProcessSorter.sortProcessNodes(
```

Comment:
Good comment and I had that thought initially.  However, it may be that our particular wildcard substitution may conflict with some other template (it's the standard github action substitutions).   The various ML commons configurations do have nested JSON "templates" in some cases.   

We're protected in this case because we only substitute if the param is actually included when calling provision, so as long as there's not both a conflicting wildcard pattern AND same-named param it won't hurt.  But if we decide to try to validate/enforce a parameter because we see the pattern it could lead to problems.

Reply:
That makes sense, I didnt consider conflicting substitutions. In this case, I think we can rely on the underlying transport actions to validate the input values fully.